shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Utxow.hs

```diff
@@ -236,8 +242,10 @@ utxoWitnessed =
         (SNothing, SNothing) -> pure ()
         (SJust mdh, SNothing) -> failBecause $ MissingTxMetaData mdh
         (SNothing, SJust md') -> failBecause $ MissingTxBodyMetaDataHash (hashMetaData md')
-        (SJust mdh, SJust md') ->
+        (SJust mdh, SJust md') -> do
           hashMetaData md' == mdh ?! ConflictingMetaDataHash mdh (hashMetaData md')
+          -- check metadata value sizes
+          _protocolVersion pp <= ProtVer 2 0 || validMetaData md' ?! InvalidMetaData
```

Comment:
@nc6 I'd like to know if you think that this is too blunt a way to handle such soft forks (this was my doing :) ). My reasoning was: 1) it's very clear, 2)  we don't need any kind of flexibility here

Reply:
My preference would be to only pass in the minor version, not the major version. This is what I was meaning the other day about relative vs absolute protocol versions. Since the minor version is within the era, and different networks may need different combinations of eras (e.g. imagine a chain that starts in the Shelley era) it would be more portable to only look at the minor version for soft forks within an era.

Reply:
Is it not weird to think that if we do not replace this transition (UTXOW) in a new era, that then we would go back to not enforcing the new restrictions? Or is the idea that we would go from `2.x` directly to `3.1`?

Reply:
I think this is reasonable, but we should perhaps extract this logic elsewhere to make it clearer. For example, we could have a `SoftForks` module with something like

```
-- | Minor revision `n` in the Shelley era. We validate supplied metadata, rather than enforcing canonical encoding.
validateMetadata :: PParams -> Bool
```

Then we could import this qualified as `SoftForks` and it would be easy to search through to find any cases where a soft fork change was made